src/elements/forms/text-field.jsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import textFieldClassNames from './text-field.css'
+
+import { classNames } from 'utils'
+
+const generateId = () =>
+  // Math.random should be unique because of its seeding algorithm.
```

Comment:
Unique IDs is not a problem for JS. There are many libraries on NPM that who'll solve this. However, I did not want to have another dependency for such small stuff that I might remove or replace later. I did this just in case and used a gist you linked.

So, you are right, I have to credit it from the code.